normandy/recipes/tests/test_exports.py

```diff
@@ -0,0 +1,246 @@
+        assert requestsmock.request_history[0].url == record_url
+        assert requestsmock.request_history[0].json() == {"data": exports.recipe_as_record(recipe)}
+        assert requestsmock.request_history[1].url == collection_url
+        mock_logger.info.assert_called_with(
+            f"Published record '{recipe.id}' for recipe {recipe.name}"
+        )
+
+    def test_unpublish_deletes_record_and_approves(self, settings, requestsmock, mock_logger):
+        """Test that requests are sent to Remote Settings on unpublish."""
+
+        settings.REMOTE_SETTINGS_URL = self.test_settings["URL"]
+        settings.REMOTE_SETTINGS_USERNAME = self.test_settings["USERNAME"]
+        settings.REMOTE_SETTINGS_PASSWORD = self.test_settings["PASSWORD"]
+
+        recipe = RecipeFactory(name="Test", approver=UserFactory())
+        collection_url = (
+            f"{settings.REMOTE_SETTINGS_URL}/buckets/{settings.REMOTE_SETTINGS_BUCKET_ID}"
+            f"/collections/{settings.REMOTE_SETTINGS_COLLECTION_ID}"
+        )
+        record_url = f"{collection_url}/records/{recipe.id}"
+        auth = f"{settings.REMOTE_SETTINGS_USERNAME}:{settings.REMOTE_SETTINGS_PASSWORD}".encode()
+        request_headers = {
+            "User-Agent": KINTO_USER_AGENT,
+            "Content-Type": "application/json",
+            "Authorization": f"Basic {base64.b64encode(auth).decode()}",
+        }
+        requestsmock.request(
+            "delete", record_url, content=b'{"data": {}}', request_headers=request_headers
+        )
+        requestsmock.request(
+            "patch", collection_url, content=b'{"data": {}}', request_headers=request_headers
+        )
+
+        remotesettings = exports.RemoteSettings()
+        remotesettings.unpublish(recipe)
+
+        assert len(requestsmock.request_history) == 2
+        assert requestsmock.request_history[0].url == record_url
+        assert requestsmock.request_history[1].url == collection_url
+        mock_logger.info.assert_called_with(
+            f"Deleted record '{recipe.id}' of recipe {recipe.name}"
+        )
+
+    def test_publish_raises_an_error_if_request_fails(self, settings, requestsmock):
+        settings.REMOTE_SETTINGS_URL = self.test_settings["URL"]
+        settings.REMOTE_SETTINGS_USERNAME = self.test_settings["USERNAME"]
+        settings.REMOTE_SETTINGS_PASSWORD = self.test_settings["PASSWORD"]
+
+        recipe = RecipeFactory(name="Test", approver=UserFactory())
+        record_url = (
+            f"{settings.REMOTE_SETTINGS_URL}/buckets/{settings.REMOTE_SETTINGS_BUCKET_ID}"
+            f"/collections/{settings.REMOTE_SETTINGS_COLLECTION_ID}"
+            f"/records/{recipe.id}"
+        )
+        requestsmock.request("put", record_url, status_code=503)
+
+        remotesettings = exports.RemoteSettings()
+        with pytest.raises(kinto_http.KintoException):
+            remotesettings.publish(recipe)
+
+        assert requestsmock.call_count == 4  # retried thrice.
```

Comment:
It might be nice to check the value of `settings.REMOTE_SETTINGS_RETRY_REQUESTS` to figure out this value, in case we change in the future. It also makes it clearer why we retry three times.